### PR TITLE
Override SourceQueue.offer to accept Tags

### PR DESCRIPTION
**Pull request by JulianGriggs**
Adds the ability for a user to specify additional tags to be attached to queue metrics

### CODE REVIEW

File: atlas-pekko/src/main/scala/com/netflix/atlas/pekko/StreamOps.scala [hunk 1]
```diff
@@ -99,23 +100,41 @@ object StreamOps extends StrictLogging {
...
+     * Add the value into the queue if there is room. Returns true if the value was successfully
+     * enqueued.
+     */
     def offer(value: T): Boolean = {
       queue.offer(value) match {
-        case QueueOfferResult.Enqueued    => enqueued.increment(); true
-        case QueueOfferResult.Dropped     => dropped.increment(); false
-        case QueueOfferResult.QueueClosed => closed.increment(); false
-        case QueueOfferResult.Failure(_)  => failed.increment(); false
+        case QueueOfferResult.Enqueued    => registry.counter(enqueuedId).increment(); true
```
Comment (brharrington) [inline atlas-pekko/src/main/scala/com/netflix/atlas/pekko/StreamOps.scala:116]: Putting data into the queue is often a hot path, adding the counter lookup for every call is probably not desirable.

**Review by brharrington — COMMENTED** [review]
What is the use-case for this change? It seems like if more detail is needed it could be done on a custom metric and limit the overhead to that use-case.

**Comment by JulianGriggs** [comment]
The items that I'm enqueuing are coming from a number of different sources, each of which reports at a different rate (in this case, each source is an Alert object). It would be helpful to add the name of the source as a dimension to the queue metrics so that when there is a big jump in say `enqueued`, we can see where that jump came from. I can add a separate custom metric to get this same information, but as I was already leveraging these metrics (in a dashboard) I figured it would be reasonable to add this functionality. But I agree that this could lead to perf issues in certain cases so feel free to reject this change.

**Comment by brharrington** [comment]
Thanks for the use-case description. Given there is an easy work around will close for now to reduce performance risk here.

**Pull request closed without merging**